test/e2e/framework/nodes.go

```diff
@@ -28,30 +28,31 @@ import (
 	"k8s.io/apimachinery/pkg/types"
 )
 
-// FindNodesByGatewayLabel finds the nodes in a given cluster by matching 'submariner.io/gateway' value.
-// Nodes with the missing label will be ignored. Note the control plane node labeled as master is ignored.
-func (f *Framework) FindNodesByGatewayLabel(cluster ClusterIndex, isGateway bool) []*v1.Node {
-	return findNodesByGatewayLabel(int(cluster), isGateway)
+// FindGatewayNodes finds nodes in a given cluster by matching 'submariner.io/gateway' value.
+func (f *Framework) FindGatewayNodes(cluster ClusterIndex) []*v1.Node {
+	return findNodesByLabel(int(cluster), GatewayLabel, true)
```

Comment:
We really don't need the `findNodesByLabel` function, we can simply use a `LabelSelector`:

```suggestion
	return return KubeClients[cluster].CoreV1().Nodes().List(context.TODO(), 
	    metav1.ListOptions{
	               LabelSelector: labels.Set{GatewayLabel: "true"}.String(),
	           })
```

Reply:
@tpantelis 

I have a question regarding your suggestion.
If we are using the `LabelSelector` instead of the `findNodesByLabel` function, it means that we are changing the return structure.

It will change from `[]*v1.Node` to `*v.NodeList`.
So in each place and test with the reference to the function will need to change.
For example:
https://github.com/submariner-io/submariner/blob/devel/test/e2e/redundancy/gateway_failover.go#L68-L70
https://github.com/submariner-io/submariner/blob/devel/test/e2e/redundancy/gateway_failover.go#L76

Will change from `gatewayNodes[0].Name` to `gatewayNodes.Items[0].Name`.

Is it acceptable for you?

Reply:
You can return NodeList.Items.

Reply:
Didn't think about it.
Thank you for the idea.

Reply:
Implemented.